src/main/java/org/jetbrains/kotlinx/lincheck/strategy/ManagedStrategyTransformer.java

```diff
@@ -45,30 +50,56 @@
-    private static final Type MANAGED_STRATEGY_HOLDER_TYPE = Type.getType(ManagedStrategyHolder.class);
+    private static final Type MANAGED_STATE_HOLDER_TYPE = Type.getType(ManagedStateHolder.class);
     private static final Type MANAGED_STRATEGY_TYPE = Type.getType(ManagedStrategy.class);
+    private static final Type LOCAL_OBJECT_MANAGER_TYPE = Type.getType(LocalObjectManager.class);
+    private static final Type RANDOM_TYPE = Type.getType(java.util.Random.class);
+    private static final Type UNSAFE_TYPE = Type.getType(Unsafe.class);
+    private static final Type UNSAFE_LOADER_TYPE = Type.getType(UnsafeHolder.class);
+    private static final Type STRING_TYPE = Type.getType(String.class);
+    private static final Type CLASS_TYPE = Type.getType(Class.class);
 
     private static final Method CURRENT_THREAD_NUMBER_METHOD = new Method("currentThreadNumber", Type.INT_TYPE, NO_ARGS);
     private static final Method BEFORE_SHARED_VARIABLE_READ_METHOD = new Method("beforeSharedVariableRead", Type.VOID_TYPE, new Type[]{Type.INT_TYPE, Type.INT_TYPE});
     private static final Method BEFORE_SHARED_VARIABLE_WRITE_METHOD = new Method("beforeSharedVariableWrite", Type.VOID_TYPE, new Type[]{Type.INT_TYPE, Type.INT_TYPE});
-    private static final Method BEFORE_LOCK_ACQUIRE_METHOD = new Method("beforeLockAcquire", Type.VOID_TYPE, new Type[]{Type.INT_TYPE, Type.INT_TYPE, OBJECT_TYPE});
-    private static final Method AFTER_LOCK_RELEASE_METHOD = new Method("afterLockRelease", Type.VOID_TYPE, new Type[]{Type.INT_TYPE, Type.INT_TYPE, OBJECT_TYPE});
+    private static final Method BEFORE_LOCK_ACQUIRE_METHOD = new Method("beforeLockAcquire", Type.BOOLEAN_TYPE, new Type[]{Type.INT_TYPE, Type.INT_TYPE, OBJECT_TYPE});
+    private static final Method BEFORE_LOCK_RELEASE_METHOD = new Method("beforeLockRelease", Type.BOOLEAN_TYPE, new Type[]{Type.INT_TYPE, Type.INT_TYPE, OBJECT_TYPE});
+    private static final Method BEFORE_WAIT_METHOD = new Method("beforeWait", Type.BOOLEAN_TYPE, new Type[]{Type.INT_TYPE, Type.INT_TYPE, OBJECT_TYPE, Type.BOOLEAN_TYPE});
+    private static final Method AFTER_NOTIFY_METHOD = new Method("afterNotify", Type.VOID_TYPE, new Type[]{Type.INT_TYPE, Type.INT_TYPE, OBJECT_TYPE, Type.BOOLEAN_TYPE});
+    private static final Method BEFORE_PARK_METHOD = new Method("beforePark", Type.BOOLEAN_TYPE, new Type[]{Type.INT_TYPE, Type.INT_TYPE, Type.BOOLEAN_TYPE});
+    private static final Method AFTER_UNPARK_METHOD = new Method("afterUnpark", Type.VOID_TYPE, new Type[]{Type.INT_TYPE, Type.INT_TYPE, OBJECT_TYPE});
+    private static final Method ENTER_IGNORED_SECTION_METHOD = new Method("enterIgnoredSection", Type.VOID_TYPE, new Type[]{Type.INT_TYPE});
+    private static final Method LEAVE_IGNORED_SECTION_METHOD = new Method("leaveIgnoredSection", Type.VOID_TYPE, new Type[]{Type.INT_TYPE});
+
+    private static final Method NEW_LOCAL_OBJECT_METHOD = new Method("newLocalObject", Type.VOID_TYPE, new Type[]{OBJECT_TYPE});
```

Comment:
It is better to reference methods/fields directly, but you have to switch to Kotlin in this case... 

Example:
```
    val m = Method.getMethod(ManagedStrategy::currentThreadNumber.javaMethod)
```